src/createNativeStackNavigator.tsx

```diff
@@ -537,7 +546,7 @@ function createStackNavigator(
 
       if (backRouteIndex > 0) {
         const newRoutes = [...state.routes];
-        newRoutes.splice(backRouteIndex, 1);
+        newRoutes.splice(backRouteIndex, dismissCount > 0 ? dismissCount : 1);
```

Comment:
and that's why here even if you pass the correct number of screens to be dismissed it starts at the wrong index (always previous screen)

Reply:
Fixed in 5e9ebd3